javatests/arcs/core/storage/DirectStoreMuxerTest.kt

```diff
@@ -99,4 +159,221 @@ class DirectStoreMuxerTest {
         }
         assertThat(callbacks).isEqualTo(1)
     }
+
+    @Test
+    fun propagatesModelUpdates_fromProxyMuxer_toDrivers() = runBlockingTest {
+        val driverProvider = MockDriverProvider()
+        DriverFactory.register(driverProvider)
+
+        val directStoreMuxer = createDirectStoreMuxer()
+
+        val callbackId = directStoreMuxer.on(ProxyCallback {})
+
+        val entityCrdtA = createPersonEntityCrdt("bob", 42)
+
+        assertThat(
+            directStoreMuxer.onProxyMessage(
+                ProxyMessage.ModelUpdate(entityCrdtA.data, id = callbackId, muxId = "a")
+            )
+        ).isTrue()
+
+        val driverA = directStoreMuxer.getEntityDriver("a")
+        val capturedA = driverA.sentData.first()
+        assertThat(capturedA.toRawEntity().singletons)
+            .containsExactly(
+                "name", "bob".toReferencable(),
+                "age", 42.toReferencable()
+            )
+        assertThat(directStoreMuxer.stores.size).isEqualTo(1)
+
+        val entityCrdtB = createPersonEntityCrdt("alice", 10)
+
+        assertThat(
+            directStoreMuxer.onProxyMessage(
+                ProxyMessage.ModelUpdate(entityCrdtB.data, id = callbackId, muxId = "b")
+            )
+        ).isTrue()
+
+        val driverB = directStoreMuxer.getEntityDriver("b")
+        val capturedB = driverB.sentData.first()
+        assertThat(capturedB.toRawEntity().singletons)
+            .containsExactly(
+                "name", "alice".toReferencable(),
+                "age", 10.toReferencable()
+            )
+        assertThat(directStoreMuxer.stores.size).isEqualTo(2)
+    }
+
+    @Test
+    fun propagatesModelUpdate_fromProxyMuxer_toDriver_toOtherProxyMuxers() = runBlockingTest {
+        val driverProvider = MockDriverProvider()
+        DriverFactory.register(driverProvider)
+
+        val directStoreMuxer = createDirectStoreMuxer()
+
+        val job = Job(coroutineContext[Job])
+
+        // proxy muxer that sends a model update to direct store muxer
+        var callbackInvoked = false
+        val callbackId1 = directStoreMuxer.on(ProxyCallback {
+            callbackInvoked = true
+        })
+
+        // proxy muxer that receives a model update from direct store muxer
+        val callbackId2 = directStoreMuxer.on(ProxyCallback {
+            assertThat(it is ProxyMessage.ModelUpdate).isTrue()
+            job.complete()
+        })
+
+        // set up store for muxId "a" and register for each proxy muxer
```

Comment:
ditto about capitalizing and ending with full stop.

Reply:
Done.